### ISSUE TITLE
As in D1 could D2 bounties gain a sourceHash to the vendor that provides them

### ISSUE THREAD

**Issue by NigelBreslaw** [Bungie-net/api#673]
This is empty in the current itemDefinition. With it we could easily sort the bounties by vendor.

**Comment by floatingatoll** [comment]
Note that bounties that don't originate from a vendor would presumably
not have this - Rat King, any of the Scrapper's bounties, etc.

**Comment by vthornheart-bng** [comment]
True!  Hmm, I could have sworn that I'd responded to this already.  I guess I didn't actually hit comment!

Just to make sure we're talking about the same property, it's sourceHashes that's on your mind?

Indeed, I think that could be a useful bit of denormalization... I just need to think through some of the implications.  The sourceHashes property in D1 was actually mapping to a DestinyRewardSourceDefinition (and still does, due to an eagerness to have that data come back which we've never had time to manually rebuild), which was an artificial construct that we created for the Armory and no longer exists (it had not just Vendors, but other sources).

Let me give it some thought before I give the absolute thumbs up on this: I would like to see us return better source data for items, but I don't want to limit sourceHash to just being Vendors and then regret it later.  I also would regret it if there ended up someday being a bounty that could be obtained by multiple vendors, rendering the property obsolete because of my assumption.  (there's nothing stopping the designers from doing this, and I've been burned on these kinds of assumptions frequently enough that it gives me pause to build this kind of assumption in)

**Comment by ArkahnX** [comment]
many items in the collections have hints as to where they can be obtained, was all of that data hand crafted? If not, maybe there is similar data lying around for bounties or other items

**Comment by vthornheart-bng** [comment]
Those hints were all hand-crafted by writers/designers - it's in game content but they have no association to the entities (activities, vendors, etc...) to which they refer unfortunately.  They're just strings to give hints to folks.  But we do have the direct relationship between bounties and items being sold in vendors: there *is* a direct relationship in that case.  I'm just thinking about whether/how I want to denormalize that data.

**Comment by ArkahnX** [comment]
wasn't there at one point in the vendorDefinitions data on all of what a vendor could sell? Would something like that work?

**Comment by vthornheart-bng** [comment]
Indeed!  That's what I was referring to when I said "But we do have the direct relationship between bounties and items being sold in vendors."

The question is whether we want to reverse that relationship for denormalization purposes - but indeed, anyone looking for this relationship could do this today, with that extra step of having to parse the Vendor definitions first!

**Comment by ArkahnX** [comment]
For the short term I think the recommendation of parsing the vendors to see if they sell the item is a passable alternative. No API calls required, just process the manifest vendorDefinitions
